src/foam/nanos/auth/CompoundAndAuthorizer.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.predicate.Predicate;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+public class CompoundAndAuthorizer implements Authorizer {
+
+  protected Authorizer[] authorizers_;
+
+  public CompoundAndAuthorizer(Authorizer[] authorizers) {
+    authorizers_ = authorizers;
+  }
+
+  // Shortcut for testing all authorizers
+  protected void consume(Consumer<Authorizer> c) throws foam.nanos.auth.AuthorizationException {
```

Comment:
I think we had this discussion before, but I don't remember the result. What is the overhead of using Streams? It used to be very high and all it is saving you from is a for loop. Also, use of streams prevents this code from working on current Android, if that ever becomes an issue.